Source/Core/Trackers/InspirationTracker.cs

```diff
@@ -8,152 +8,62 @@ namespace PrisonLabor.Core.Trackers
+            if (prisonersCount == 0)
             {
-                Log.Message("PrisonLabor Warning: InspirationTracker.GetInsipirationValue() didn't find map or pawn. Maybe it was called before calculating values.");
-                return 0;
+                isWatched[pawn] = 0.0f;
+                return false;
             }
-            lock (inspirationValues)
+
+            if (wardensCount == 0)
             {
-                return inspirationValues[map][pawn];
+                isWatched[pawn] = -0.00425f;
+                return false;
             }
-        }
 
-        public static void Calculate(Map map)
-        {
-            if (InspirationTracker.tickCounter >= InspirationTracker.updateInterval)
+            if (room.IsHuge || room.CellCount > 1600)
             {
-                if (map.mapPawns.PrisonersOfColonyCount == 0)
-                {
-                    return;
-                }
-
-                InspirationTracker.tickCounter = 0;
-
-                lock (inspirationValues)
-                {
-
-                    var wardens = new List<Pawn>();
-                    wardens.AddRange(map.mapPawns.FreeColonists);
-
-                    var prisoners = new List<Pawn>();
-                    prisoners.AddRange(map.mapPawns.PrisonersOfColony);
-
-                    if (wardens.Count == 0 || prisoners.Count == 0)
-                    {
-                        return;
-                    }
-                    
-                    Dictionary<Pawn, float> mapCalculations;
-                    if (inspirationValues.TryGetValue(map, out mapCalculations))
-                    {
-                        mapCalculations.Clear();
-                    }
-                    else
-                    {
-                        inspirationValues[map] = mapCalculations = new Dictionary<Pawn, float>();
-                    }
-
-                    foreach (var prisoner in prisoners)
-                    {
-                        mapCalculations[prisoner] = 0f;
-                    }
-
-                    var roomsIdtoWardens = new Dictionary<int, List<Pawn>>();
-                    var wardenCurrentPrisoners = new Dictionary<Pawn, List<Pawn>>();
-
-                    foreach (var warden in wardens)
-                    {
-                        if(warden.GetRoom() == null)
-                        {
-                            continue;
-                        }
-
-                        var roomId = warden.GetRoom().ID;
-
-                        if (!roomsIdtoWardens.ContainsKey(roomId))
-                        {
-                            roomsIdtoWardens[roomId] = new List<Pawn>();
-                        }
+                isWatched[pawn] = -0.005f;
+                return false;
+            }
 
-                        roomsIdtoWardens[roomId].Add(warden);
-                    }
+            if (prisonersCount / wardensCount > 4f)
```

Comment:
I think that amount should be put in mod option. I mean how many prisoners can handle single warden.

Reply:
Lol, ok however i suck at GUI

Reply:
Nothing fancy, simple slider with 2 to 8 values :D

Reply:
<img width="901" alt="Screen Shot 2020-07-06 at 7 56 08 PM" src="https://user-images.githubusercontent.com/6241020/86697049-dff29880-bfc2-11ea-9b13-1462112df895.png">

Reply:
I'll remove the warning

Reply:
<img width="916" alt="Screen Shot 2020-07-06 at 8 03 09 PM" src="https://user-images.githubusercontent.com/6241020/86697956-b8500000-bfc3-11ea-9cac-59f9d8f0d0b2.png">